front/app/utils/cl-router/history.ts

```diff
@@ -1,29 +1,37 @@
 import { first } from 'rxjs/operators';
 import { localeStream } from 'utils/locale';
 import updateLocationDescriptor from 'utils/cl-router/updateLocationDescriptor';
+import { scrollToTop as scrollTop } from 'utils/scroll';
 // tslint:disable-next-line:no-vanilla-routing
 
 import history from 'utils/browserHistory';
 
 // overrides push and replace methods so they update the location with the current locale from the locale stream
 function historyMethod(
   method: 'push' | 'replace',
-  location: Partial<Location> | string
+  location: Partial<Location> | string,
+  scrollToTop?: boolean
```

Comment:
Let's make this an object where one of the keys is `scrollToTop`. This will make it easier if we want to add new settings in the future + would make the use of this setting much more explicit. So instead of `clHistory.push('/', true)` you would have `clHistory.push('/', { scrollToTop: true })`

Reply:
@IvaKop Ok sounds good - yeah much more clear that way :+1: Do you mean something like this then? Or do we have a different preferred format? :thinking: 

```
type ClHistoryProps = {
  scrollToTop?: boolean;
};

// overrides push and replace methods so they update the location with the current locale from the locale stream
function historyMethod(
  method: 'push' | 'replace',
  location: Partial<Location> | string,
  props?: ClHistoryProps
): void {
  // 'gets' current locale
  localeStream()
    .observable.pipe(first())
    .subscribe((locale) => {
      // calls the vanilla react-router method with updated location
      history[method](updateLocationDescriptor(location, locale));

      if (props?.scrollToTop) {
        scrollTop();
      }
    });
}

export default {
  ...history,
  push: (location: Partial<Location> | string, props?: ClHistoryProps): void =>
    historyMethod('push', location, { scrollToTop: props?.scrollToTop }),
  replace: (
    location: Partial<Location> | string,
    props?: ClHistoryProps
  ): void =>
    historyMethod('replace', location, { scrollToTop: props?.scrollToTop }),
  goBack: () => history.back(),
};
```

Reply:
Yes, exactly! Although I would probably name this `options` instead of props as I associate props more with a React component.

Reply:
Ahh true - ok I'll change the name :) Thanks Iva!